pgjdbc/src/test/java/org/postgresql/test/jdbc2/DriverTest.java

```diff
@@ -104,6 +104,7 @@ public void testAcceptsURL() throws Exception {
       verifyUrl(drv, "jdbc:postgresql://?service=driverTestService1", "test-host1", "5444", "testdb1");
       verifyUrl(drv, "jdbc:postgresql://?service=driverTestService1&host=other-host", "other-host", "5444", "testdb1");
       verifyUrl(drv, "jdbc:postgresql:///?service=driverTestService1", "test-host1", "5444", "testdb1");
+      verifyUrl(drv, "jdbc:postgresql:///somedb?service=driverTestService1", "test-host1", "5444", "somedb");
```

Comment:
@davecramer
forgive me if this is not good place for that comment

I have more test cases which possibly does not work as libpq (or is not there as test case)
but current `verifyUrl()` does not allow to test them.

For example the following url is also valid:
```
jdbc:postgresql://user@host:port/db_name?service=service_name
```

Would it be fine to extend `verifyUrl()` like this?
```
private void verifyUrl(Driver drv, String url, String hosts, String ports, String dbName
, String username, String password)
```

Reply:
@MarekUniq this is fine. 

Ya, this url is also valid. Let me finish my PR and then you can add another one. Or simply add another one to fix it and I will sort it out.

Please continue though. This thing is pretty complicated it seems, so the more eyes the better.

Reply:
@davecramer
I rather wait for your PR. I need to make experiments with libpq as first step.

Reply:
@davecramer - I have analyzed libpq behaivour and tried to implement it for pgjdbc.
I feel that current parser code is not well readable. I will try to make it more readable.
I try to come up with PR within a week or two.
If issue #2393 and PR #2395 is not urgent then I think you can stop working on them because my work will implement fix anyway.

Reply:
> @davecramer - I have analyzed libpq behaivour and tried to implement it for pgjdbc. I feel that current parser code is not well readable. I will try to make it more readable. I try to come up with PR within a week or two. If issue #2393 and PR #2395 is not urgent then I think you can stop working on them because my work will implement fix anyway.

I can wait. I don't think this is urgent